src/App.js

```diff
@@ -259,13 +283,43 @@ class App extends React.Component {
     }, 1000)
   }
 
+  handleIdentityCancel = () => {
+    this.setState({ identityIntent: null })
+  }
+
+  handleIdentitySave = ({ address, label }) => {
+    const { resolve, reject } = this.state.identityIntent
+    this.state.wrapper
+      .modifyAddressIdentity(address, { name: label })
+      .then(() => this.setState({ identityIntent: null }, resolve))
+      .catch(reject)
+  }
+
+  handleIdentityResolve = address => {
+    // returns promise
+    if (this.state.wrapper) {
+      return this.state.wrapper.resolveAddressIdentity(address)
+    } else {
+      // wrapper has not been initialized
+      // re-request in 100 ms
+      return new Promise(resolve => {
```

Comment:
Under what circumstances does this happen?  Perhaps it would be good to add a comment about this.

Is it:
- the wrapper hasn't been initialised
- there are addresses asking to be resolved in native apps (if the wrapper isn't initialised iframes can't request resolution)

Reply:
Put your hands together, it happens when a native app (in case it is rendering) tries to resolve an address before the wrapper has been initialised.